docs/ref/models/instances.txt

```diff
@@ -474,10 +474,17 @@ follows this algorithm:
 * If the object's primary key attribute is *not* set or if the ``UPDATE``
   didn't update anything, Django executes an ``INSERT``.
 
-The one gotcha here is that you should be careful not to specify a primary-key
-value explicitly when saving new objects, if you cannot guarantee the
-primary-key value is unused. For more on this nuance, see `Explicitly specifying
-auto-primary-key values`_ above and `Forcing an INSERT or UPDATE`_ below.
+There are a few gotchas and exceptions to the above logic:
+
+* Given that :attr:`~django.db.models.Field.primary_key` is read-only,
+  calling ``save`` on an instance for which you've modified the existing
+  primary key will *always* result in an ``INSERT`` being executed,
+  regardless of the two conditions above.
+* You should be careful not to specify a primary-key
+  value explicitly when saving new objects, if you cannot guarantee the
+  primary-key value is unused. For more on this nuance, see `Explicitly
+  specifying auto-primary-key values`_ above and `Forcing an INSERT or UPDATE`_
+  below.
```

Comment:
I would rather add an example to the algorithm description, because the second case
> _"if the ``UPDATE`` didn't update anything, Django executes an ``INSERT``..."_

describes what you want to clarify. When you manually change primary key to an existing one then Django updates that objects and **doesn't create** a new one. On the other hand if you change it to non-existing primary key then Django creates a new instance, so proposed sentence _"will *always* result in an ``INSERT``"_ is not true.

Maybe

> * If the object's primary key attribute is *not* set or if the ``UPDATE`` 
>   didn't update anything (**e.g. ...**), Django executes an ``INSERT``.

Reply:
Okay, thanks for clarifying @felixxm.  Will revise this PR when I have a moment this week.